app/controllers/posts_controller.rb

```diff
@@ -6,15 +6,16 @@ class PostsController < ApplicationController
   # GET /posts
   # GET /posts.json
   def index
-    @posts =  Post.all.order(created_at: :desc).collect do |post|
+    @posts =  Post.where(user_id: current_user.id).all.order(created_at: :desc).collect do |post|
                 { 
                   id: post.id,
                   title: post[:title],
                   content: post[:content].split(" ").first(20).join(" "),
                   email: post.user.email,
-                  created_at: post.created_at 
-                }                  
+                  created_at: post.created_at,
+                }                                
               end
+    @users= User.where(id: current_user.id)
```

Comment:
We can do a slight refactoring here (i know i wrote the code). Remove this query from here and move to model. So it will be like this.

```ruby
  # in controller
  def index
     @post  = Post.posts
     @users = Array(current_user)
  end
```

```ruby
  # in Post model
  def self.my_post(current_user)
    where(user: current_user)
      .order(created_at: :desc)
      .collect do |post|
         { .. }
      end
  end
```

Reply:
made the change as suggested